### PR TITLE
h5 and h6

### PR DESCRIPTION
**Pull request by dr-programmer**

### CODE REVIEW

**Comment by dr-programmer** [comment]
@VladislavVulchevChaos

**Comment by dr-programmer** [comment]
@VladislavVulchevChaos when you can please review this pull request, because i am ready with h7 and have to wait for this pull request to be merged. Thanks!

File: SourceCode/h5/triangle2.cpp [hunk 1]
```diff
@@ -0,0 +1,226 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <cmath>
+
+class Vector {
+    float x, y, z;
```
Comment (VladislavVulchevChaos) [inline SourceCode/h5/triangle2.cpp:7]: It's good to be able to index into the vertices, so storing them in an array could be better.

File: SourceCode/h5/triangle2.cpp [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    }
+
+    float dotProduct(const Vector a, const Vector b) {
+        return a.getX() * b.getX() 
+                + a.getY() * b.getY() 
+                + a.getZ() * b.getZ();
+    }
+};
+
+#define NUM_OF_TRIANGLE_VERTICES 3
```
Comment (VladislavVulchevChaos) [inline SourceCode/h5/triangle2.cpp:86]: You could use constexpr instead of #defines. It's safer.

File: SourceCode/h5/triangle2.cpp [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+    void cacheNormal() {
+        normal = calculateNormal();
+        normal.normalize();
+    }
+    void cacheNormal(const Vector v) {
+        normal = v;
+    }
+    
+    Vector getVertex0() const {
```
Comment (VladislavVulchevChaos) [inline SourceCode/h5/triangle2.cpp:110]: Only 1 getVertex(const int idx) is enough

File: SourceCode/h5/triangle2.cpp [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+        return vertex[2];
+    }
+    Vector getVertex(const unsigned int index) const {
+        return vertex[index];
+    }
+    Vector getNormal() const {
+        return normal;
+    }
+
+    void setVertex0(const Vector v) {
```
Comment (VladislavVulchevChaos) [inline SourceCode/h5/triangle2.cpp:126]: Only one setVertex(idx, value) could be used

File: SourceCode/h5/triangle2.cpp [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+    void setVertex(const unsigned int index, const Vector v) {
+        vertex[index] = v;
+    }
+};
+
+static const int imageWidth = 1920;
+static const int imageHeight = 1080;
+static const int maxColorComponent = 255;
+
+#define SET_COLORS(COLORS, TR_INDEX)    for(unsigned int i = 0; i < 3; i++) { \
```
Comment (VladislavVulchevChaos) [inline SourceCode/h5/triangle2.cpp:144]: As a whole using complex logic in #defines is bad practice

File: SourceCode/h5/triangle2.cpp [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+                                            COLORS[i] = ((TR_INDEX + 1) * (i + 16) * 64 \
+                                                            + (i+1) * 64) % 256; \
+                                        }
+
+#define CLEAR_COLORS(COLORS)    for(unsigned int i = 0; i < 4; i++) { \
+                                    COLORS[i] = 0; \
+                                }
+
+int main() {
```
Comment (VladislavVulchevChaos) [inline SourceCode/h5/triangle2.cpp:153]: Decouple the logic here into dedicated files and structures

**Pull request merged**